ml-proto/spec/int.ml

```diff
@@ -208,7 +208,55 @@ struct
   let ge_s x y = x >= y
   let ge_u x y = cmp_u x (>=) y
 
-  let of_string = Rep.of_string
+  (* This implementation allows leading signs and unsigned values *)
+  let of_string x =
+    let len = String.length x in
+    let power_of_two n = Rep.shift_left Rep.one n in
+    let ten = Rep.of_int 10 in
+    let fail () = failwith "int_of_string" in
```

Comment:
You could make this more compact by turning it into a `require` function, similar e.g. to that used in `check.ml`.

The string should probably be `"of_string"`.